spec/services/recurring_contribution/subscriptions/juntos_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+RSpec.describe RecurringContribution::Subscriptions::Juntos do
+  include RecurringContributionsHelper
+
+  describe '#process' do
+    let(:pagarme_plan) { create(:plan) }
+    let!(:juntos_plan) { create(:plan, plan_code: pagarme_plan.id) }
+    let(:project) { create(:project) }
+    let(:user) { create(:user) }
+    let(:pagarme_subscription) { build_subscription_mock('credit_card', pagarme_plan) }
+
+    context 'when the models subscription and transaction are successfully created' do
+      let(:juntos_subscription_data) {
+         RecurringContribution::Subscriptions::JuntosSubscriptionData.new(user, 'credit_card', pagarme_subscription)
+      }
+      let(:juntos_subscription_service) {
+        RecurringContribution::Subscriptions::Juntos.new(project, juntos_subscription_data)
+      }
+      let(:juntos_subscription) { juntos_subscription_service.process }
+      let(:juntos_transaction) { juntos_subscription.transactions.first }
+
+      it 'must create a subscription model' do
+        expect(juntos_subscription.id).to_not be_nil
+      end
+
+      it 'must create a transaction model' do
+        expect(juntos_transaction.id).to_not be_nil
+      end
+
+      it "must create a subscription model with subscription_code equal to pagarme's subscription id" do
+        expect(juntos_subscription.subscription_code).to eq(pagarme_subscription.id)
+      end
+
+      it "must create a transaction model with transaction_code equal to pagarme's transaction id" do
+        expect(juntos_transaction.transaction_code).to eq(pagarme_subscription.current_transaction.id)
+      end
+    end
+
+    context 'when a problem occurs on creation of models subscription and transaction' do
+      let(:juntos_subscription_data) {
+         RecurringContribution::Subscriptions::JuntosSubscriptionData.new(user, 'invalid_payment_method', pagarme_subscription)
+      }
+      let(:juntos_subscription_service) {
+        RecurringContribution::Subscriptions::Juntos.new(project, juntos_subscription_data)
+      }
+
+      before do
+        juntos_subscription_service.process
+      end
+
+      it 'should not create a subscription model' do
+        expect(Subscription.count).to eq(0)
```

Comment:
Esquema massa para ver se uma classe altera a contagem assim

```ruby
expect { juntos_subscription_service.process }.to change { Subscription.count }.by(0)
```

Tem o `to_not`, mas tentei usar ele em um projeot que eu to, e deu uns erros estranhos. Ve se funciona aq.
Mais informação: https://www.relishapp.com/rspec/rspec-expectations/v/2-2/docs/matchers/expect-change